lib/api_me/sorting.rb

```diff
@@ -26,20 +31,22 @@ def sort_meta
 
     protected
 
-    def sort
-      unless sort_criteria === ""
-        sort_p = sort_criteria
+    def sort(criteria = default_sort_criteria)
+      unless sort_association == ""
+        criteria_class = criteria.camelize.constantize
+
         if sort_reverse === "true"
-          self.scope = scope.sort_by {|scope| scope[sort_p]}.reverse!
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :desc))
         else
-          self.scope = scope.sort_by {|scope| scope[sort_p]}
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :asc))
```

Comment:
In fact, wouldn't you usually want to still do a left outer join here for your sorting? At least I believe the feature for which is feature is requested probably needs a left join instead of an inner join.